File: cmps101/pa2/Lex.c
/*Tim Mertogul
* 1331402
* CMPS101 Summer14
* 07/09/14
* Lex reads from input file and alphabetices
* Lex.c
*/


#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include "List.h"

#define MAX_LEN 200

int main(int argc, char * argv[]){

   int linecount = 0;  //n, count=0,
   FILE *in, *out;
   char line[MAX_LEN];
   List Lu = newList();
   //List Ls = newList();

   // check command line for correct number of arguments
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(1);
   }
   //open file, count tokens, close file
   in = fopen(argv[1], "r");
   out = fopen(argv[2], "w");
   if( in==NULL ){
      printf("Unable to open file %s for reading\n", argv[1]);
      exit(1);
   }
   if( out==NULL ){
      printf("Unable to open file %s for writing\n", argv[2]);
      exit(1);
   }
   while( fgets(line, MAX_LEN, in) != NULL)  {
	  linecount++;
   }
   //printf("linecount = %d \n", linecount);

   rewind(in);
   //char Lu[linecount] [MAX_LEN];
   char words[linecount] [MAX_LEN];
   char copy[linecount] [MAX_LEN];

   for (int i = 0; i < linecount; i++){
	  fgets(line,MAX_LEN,in);
	  strcpy(words[i], line);
	  strcpy(copy[i], line);
   }
/*
for (int j=0; j<linecount; j++){
   fprintf(out, "%s\n", words[j]);
}  */ 


	for(int i=0; i<linecount; i++){
		append(Lu,i);
	}

   //printList(out, Lu);

   /* close files */
   fclose(in);
   in = NULL;

   for(int x=1; x<linecount; x++){
		//String tempy = words[x];
		char tempy[MAX_LEN]; // = words[x];
		strcpy(tempy, words[x]);

		moveTo(Lu,x);
		//int Pt = getIndex(Lu);
		int Ti;
		int jk;
		//for (jk=x-1; jk>=0 && tempy.compareTo(words[jk])<0; jk--){
		for (jk=x-1; jk>=0 && strcmp(tempy, words[jk]) <0; jk--){
			moveTo(Lu,x);
			Ti = getElement(Lu);
			delete(Lu);
			moveTo(Lu,jk);
			insertBefore(Lu,Ti);
			x--;
			strcpy(words[jk+1], words[jk]);
			//words[jk+1] = words[jk];
		}
		if(getIndex(Lu) >= 0){
			movePrev(Lu);
		}
		strcpy(words[jk+1], tempy);
		//words[jk+1] = tempy;
		//free(tempy);
	}
	int Ti = 0;
	/*while (Ti < words.length ){
		System.out.println("a[T] = "+words[Ti]);
		Ti++;
	}
	Ti = 0;*/
	while (Ti < linecount ){
		moveTo(Lu,Ti);
		fprintf(out, "%s", copy[ getElement(Lu) ]);
		Ti++;
	}
	moveTo(Lu,0);


   fclose(out);
   out=NULL;

   return(0);
}

File: cmps101/pa1/Lex.java
/*Tim Mertogul
* 1331402
* CMPS101 Summer14
* 07/02/14
* ADD DESCRIPTION
* Lex.java
*/

import java.io.*;
import java.util.Scanner;


class Lex {
	
	 /*static List InsertionSort(List Lu, String[] a) {
				
		//System.out.print(Lu);
		return Lu;
	}*/
		 
	public static void main(String[] args) throws IOException{
		Scanner Lin = null;
		Scanner in = null;
		PrintWriter out = null;
		String line = null;
		//String[] elements = null;
		List Lu = new List();
		List Ls = new List();
	    int i, j, n=0;

		if(args.length != 2){
			throw new IOException("Usage: Lex input output");
			//System.exit(1);
		}
		Lin = new Scanner(new File(args[0]));
		int linecount = 0;		
		while ( Lin.hasNext() ) {
			Lin.next();
			linecount++;
		}
		Lin.close();
		in = new Scanner(new File(args[0]));
		out = new PrintWriter(new FileWriter(args[1]));
		//L=0;
		String[] words = new String[linecount];
		String[] copy = new String[linecount];

		int temp = 0;
		while( in.hasNextLine() ){
			line = in.nextLine();
			copy[n] = line;
			words[n] = line;
			n++;
		}
		int T = 0;
		/*System.out.println("words.length = "+words.length);
		while (T < words.length ){
			System.out.println("words[T] = "+words[T]);
			T++;
			
		}*/
		in.close();
		
		for(i=0; i<linecount; i++){
			Lu.append(i);
		}
		//System.out.println("unsorted indices = "+Lu);
		

		//Ls = InsertionSort(Lu, words);
		//out.println(Ls);
		//Ls.moveTo(0);
		
		//new String[] a = words;
		//new String[words.length()] copy;
		//copy = words;

		for(int x=1; x<words.length; x++){
					String tempy = words[x];
					Lu.moveTo(x);
					int Pt = Lu.getIndex();
					int Ti;
					int jk;
					for (jk=x-1; jk>=0 && tempy.compareTo(words[jk])<0; jk--){
						//System.out.println("Lu = "+Lu);
						//System.out.println("getIndexB"+Lu.getIndex());

						//System.out.println("i="+i);
						//System.out.println("j="+j);
						//Lu
						Lu.moveTo(x);
						Ti = Lu.getElement();
						//System.out.println("getIndexDELETE"+Lu.getIndex());
						Lu.delete();
						//System.out.println("getIndexj = "+j);
						//System.out.println("getLENGTHj = "+Lu.length());
						Lu.moveTo(jk);
						Lu.insertBefore(Ti);
						//System.out.println("getIndexE"+Lu.getIndex());
						x--;
						words[jk+1] = words[jk];
					}
					if(Lu.getIndex() >= 0){
						Lu.movePrev();
					}
					words[jk+1] = tempy;
		}
		int Ti = 0;
		/*while (Ti < words.length ){
			System.out.println("a[T] = "+words[Ti]);
			Ti++;
		}
		Ti = 0;*/
		while (Ti < copy.length ){
			Lu.moveTo(Ti);
			out.println(copy[ Lu.getElement() ]);
			Ti++;
		}
		//copy = a;
		Lu.moveTo(0);
		
		/*for(Lu.getIndex(); Lu.getIndex() < Lu.length(); Lu.moveNext()){
			out.println(copy[Lu.getElement()]);
		}*/
		
		out.close();

		
	}
		
}

//public static String[] InsertionSort(String[] toList, int length){
	//String[] Sorted = new String[length]; 
	
	
	/*for (int i=1; i < toList.length; i++){
		String tem = toList[i];
		int j;
		for (j=i-1; j>=0 && tem<toList[j]; j--)
			toList[j+1] = toList[j];
		toList[j+1] = tem;
		
	}*/
	//return toList;
//}